RT_UPDATE/download_nLx.R

```diff
@@ -0,0 +1,416 @@
+#' nLx_data <- download_nLx(location = "Germany",
+#'                          gender = "female",
+#'                          nLxDatesIn = c(2000, 2005, 2010),
+#'                          output = "abridged")
+#'
+#' # Example using user-supplied matrix
+#' nLx_data <- matrix(runif(20), nrow = 4)
+#' download_nLx(nLx = nLx_data,
+#'              nLxDatesIn = c(2000, 2005, 2010, 2015),
+#'              output = "5-year")
+#' }
+#'
+#' @export
+
+# updated to handle 5-year output ot single year output (age)
+download_nLx <- function(nLx        = NULL,
+                         location   = NULL,
+                         gender     = NULL,
+                         nLxDatesIn = NULL,
+                         method     = "linear",
+                         output     = "5-year",
+                         radix      = 1,
+                         ...) {
+  
+  verbose <- getOption("basepop_verbose", TRUE)
+  
+  # if nLx is provided by user just return back the nLx data
+  # we assume user provides data as vector or as a matrix
+  if(!is.null(nLx)) {
+    
+    nLx           <- as.matrix(nLx)
+    colnames(nLx) <- nLxDatesIn
+    n             <- nrow(nLx)
+    
+    
+    # this part here deals with the output variation in age
+    # user can choose single, abridged or 5 year
+    # based on the user choice, the rownames are correspondingly assigned
+    # using nrow of nLx
+    if(output == "single") { 
+      
+      Age <- 0:(n - 1)
+      
+    }
+    
+    if(output == "abridged") {
+      
+      Age <- c(0, 1, seq(5, (n - 2) * 5, by = 5))
+      
+    } 
+    
+    if(output == "5-year") {
+      
+      Age <- seq(from = 0, length.out = n, by = 5)
+      
+    }
+    
+    rownames(nLx) <- Age
+    
+    return(nLx)
+    
+  }
+  
+  # this part goes until the end of function and deals with missing nLx
+  if(is.null(nLx)) {
+    
+    # At the very least we need a location data. 
+    # if no location data is provided we stop with error
+    if(is.null(location)) {
+      
+      stop("You need to provide a location to download the data for nLx")
+      
+    }
+    
+    # I check which wpp versions are available on the machine
+    # if no wpp is available we stop and offer to use our function to download one
+    # otherwise I use the latest available wpp on the machine (in case > 1) for the mortality data 
+    
+    # installed wpp versions
+    installed_wpp <- grep("^wpp\\d{4}$", rownames(installed.packages()), value = TRUE)
+    
+    # stop if none
+    if(length(installed_wpp) == 0) {
+      
+      stop("No wpp package installed. Please use function install_wpp to download the wpp package.")
+      
+    }
+    
+    # find the lates one
+    latest_wpp <- sort(installed_wpp, decreasing = TRUE)[1]
+    
+    # UPDATE:
+    # we basically want wpp to be at leat 2022. We need this for single year data.
+    # Since we have a dedicated wpp install function, we agreed that this is ok to do
+    if(latest_wpp < 2022) {
```

Comment:
same story: use the 5-y data and interpolate it in this case, with message/warning including the call required to update wpp.